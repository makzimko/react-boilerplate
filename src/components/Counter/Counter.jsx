import React from 'react';
import { compose } from 'redux';
import PropTypes from 'prop-types';

import { withBehavior, withSelector } from '../../enhancers';
import counterBehavior from './behavior';
import counterSelector from './selector';

const propTypes = {
  count: PropTypes.number,
  onDecreaseClick: PropTypes.func,
  onIncreaseClick: PropTypes.func,
};

const defaultProps = {
  count: 0,
  onDecreaseClick: Function.prototype,
  onIncreaseClick: Function.prototype,
};

const Counter = React.memo((props) => {
  const { count, onDecreaseClick, onIncreaseClick } = props;
  return (
    <div>
      <h3> Counter </h3>
      <button type="button" onClick={onDecreaseClick}>
        -
      </button>
      <span>{count}</span>
      <button type="button" onClick={onIncreaseClick}>
        +
      </button>
    </div>
  );
});

Counter.propTypes = propTypes;
Counter.defaultProps = defaultProps;

export default compose(
  withSelector(counterSelector),
  withBehavior(counterBehavior),
)(Counter);
