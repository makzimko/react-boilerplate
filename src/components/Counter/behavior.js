import { counter } from '../../actions';

const onDecreaseClick = (dispatch) => () => {
  dispatch(counter.decrease());
};

const onIncreaseClick = (dispatch) => () => {
  dispatch(counter.increase());
};

export default {
  onDecreaseClick,
  onIncreaseClick,
};
