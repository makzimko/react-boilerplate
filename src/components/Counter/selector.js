import { createStructuredSelector } from 'reselect';

const counterSelector = (state) => state.counter;

export default createStructuredSelector({
  count: counterSelector,
});
