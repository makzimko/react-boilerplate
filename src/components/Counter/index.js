// eslint-disable-next-line
import Counter from './Counter';

export default Counter;
