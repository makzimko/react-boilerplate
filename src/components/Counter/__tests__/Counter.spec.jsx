import React from 'react';

import { Counter } from '../Counter';

describe('Counter component', () => {
  it('should do something', () => {
    const component = shallow(<Counter />);

    expect(component).toMatchSnapshot();
  });
});
