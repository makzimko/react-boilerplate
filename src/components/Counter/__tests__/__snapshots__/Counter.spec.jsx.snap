// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should do something 1`] = `
<div>
  <h3>
     Counter 
  </h3>
  <button
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
    type="button"
  >
    +
  </button>
</div>
`;
